src/communication/xcWebSocketSubscriber.ts

```diff
@@ -132,26 +135,20 @@ export class DefaultSubscriber implements Subscriber {
     };
 
 
-    getSnapshot(componentName: string, stateMachineName: string, getSnapshotListener: (items: Array<Packet>) => void): void {
-        let replyTopic = uuid();
-        let thisSubscriber = this;
-        this.observableMsg
-            .map((rawMessage: MessageEvent) => thisSubscriber.deserializer.deserialize(rawMessage.data || rawMessage))
-            .filter((data: DeserializedData) => (data.command === Commands[Commands.snapshot] && data.topic === replyTopic))
-            .subscribe((data: DeserializedData) => {
-                thisSubscriber.sendUnsubscribeRequestToTopic(replyTopic, Kinds.Snapshot);
-                getSnapshotListener(thisSubscriber.getJsonDataFromSnapshot(data.stringData, data.topic));
-            });
-        this.sendSubscribeRequestToTopic(replyTopic, Kinds.Snapshot);
-        let dataToSendSnapshot = this.getDataToSendSnapshot(componentName, stateMachineName, replyTopic);
-        this.webSocket.send(thisSubscriber.serializer.convertToWebsocketInputFormat(dataToSendSnapshot));
-    };
-
-    getSnapshotPromise(componentName: string, stateMachineName: string): Promise<Array<Packet>> {
+    getSnapshot(componentName: string, stateMachineName: string): Promise<Array<Packet>> {
```

Comment:
Perhaps it is a good time to find another name of the Packet type
@maurelio1234  no ?